litmus/preSi.ml

```diff
@@ -1607,53 +1616,52 @@ module Make
                 (pctag (proc,addr)) addr)
             addrs
         end ;
-        let mem_map =
-          let open BellInfo in
-          match test.T.bellinfo with
-          | None|Some { regions=None;_} -> []
-          | Some { regions=Some map;_} -> map in
-        begin match pte_init,mem_map with
-        | [],[] -> ()
-        | bds,_ ->
+        begin match pte_init with
+        | [] -> ()
+        | bds ->
             O.oii "barrier_wait(_b);" ;
             List.iter
               (fun x ->
-                let ok1 = try
+                try
                   begin match Misc.Simple.assoc x bds with
                   | P phy ->
                       O.fii
                         "(void)litmus_set_pte_safe(%s,_vars->pte_%s,_vars->saved_pte_%s);"
-                        x x phy
+                        x x phy ;
+                      O.fii "litmus_flush_tlb((void *)%s);" x
                   | Z ->
-                      O.fii "(void)litmus_set_pte_safe(%s,_vars->pte_%s,litmus_set_pte_invalid(*_vars->pte_%s));" x x x
+                      O.fii "(void)litmus_set_pte(%s,_vars->pte_%s,litmus_set_pte_invalid(*_vars->pte_%s));" x x x ;
+                      O.fii "litmus_flush_tlb((void *)%s);" x
                   | V (o,pteval) ->
```

Comment:
@maranget, I am not sure why we define this type:
```
type pte_addr = V of string option * A.V.PteVal.t | P of string | Z
```
and use it here? I imagine there was a time where we could say `[PTE(x)]=0;` or `[PTE(x)]=[PTE(y)];`? Neither of them is supported in the parser currently? Can we simplify it and keep the V of string option * A.V.PteVal.t only?